llama-index-server/llama_index/server/utils/agent_tool.py

```
@@ -0,0 +1,230 @@
+            ctx,
+            tool,
+            tool_call,
+            event_emitter=lambda msg: ctx.write_event_to_stream(
+                AgentRunEvent(
+                    name=agent_name,
+                    msg=msg,
+                    event_type=AgentRunEventType.PROGRESS,
+                    data={
+                        "id": progress_id,
+                        "total": total_steps,
+                        "current": i,
```

Ruff (0.8.2)

Function definition does not bind loop variable `i`

(B023)